### PR TITLE
Add validate_embed

### LINKED ISSUE #9: Add util method to check validity of an Embed object
### Summary

Embeds have a character limit for various fields and the embed itself collectively, 
> - Embed title is limited to 256 characters
> - Embed description is limited to 4096 characters
> - An embed can contain a maximum of 25 fields
> - A field name/title is limited to 256 character and the value of the field is limited to 1024 characters
> - Embed footer is limited to 2048 characters
> - Embed author name is limited to 256 characters
> - The total of characters allowed in an embed is 6000

having a function that checks if an Embed is send-able can be handy. 

### Why is this needed?

Invalid Form Body requests can be avoided with a proper check before sending the embed.

### Ideal implementation

Returning a bool object would be ideal.

```py
def check_embed_limit(embed: hikari.Embed) -> bool: 
    if <exceeds any limit>: # multiple checks like this...
         return False
    return True
```

### Checklist

- [X] I have searched the issue tracker and have made sure it's not a duplicate. If it is a follow up of another issue, I have specified it.

### PR DESCRIPTION
**Pull request by HyperGH**
Add `validate_embed()` which validates all field lengths of an embed.

Closes #9

### CODE REVIEW

File: toolbox/utils.py [hunk 1]
```diff
@@ -464,6 +466,55 @@ def as_command_choices(*args: t.Any, **kwargs: t.Any) -> t.Sequence[hikari.Comma
...
+            if (length := len(field.name)) > 256:
+                raise EmbedValidationError(f"Embed field name must be less than 256 characters, got {length}.")
+            if (length := len(field.value)) > 1024:
+                raise EmbedValidationError(f"Embed field value must be less than 1024 characters, got {length}.")
+
+    if embed.author and embed.author.name and (length := len(embed.author.name)) > 256:
+        raise EmbedValidationError(f"Embed author name must be less than 256 characters, got {length}.")
+
+    if embed.footer and embed.footer.text and (length := len(embed.footer.text)) > 2048:
+        raise EmbedValidationError(f"Embed footer text must be less than 2048 characters, got {length}.")
```
Comment (parafoxia) [inline toolbox/utils.py:513]: These are duplicate validations.

Comment (Lunarmagpie) [reply toolbox/utils.py:513]: I dont think they are

File: toolbox/errors.py [hunk 2]
```diff
@@ -15,6 +15,12 @@ class CacheFailureError(ToolboxError):
     ...
 
 
+class EmbedValidationError(ToolboxError):
+    """Exception raised when a embed validation fails."""
+
+    ...
```
Comment (Lunarmagpie) [inline toolbox/errors.py:21]: ... isn't needed

File: toolbox/utils.py (hunk 1 above)
Comment (parafoxia) [reply toolbox/utils.py:513]: I can't see any differences between lines 493-4 and 512-3 (same with 496-7 and 509-10), unless I'm missing something?

File: toolbox/utils.py [hunk 3]
```diff
@@ -464,6 +466,55 @@ def as_command_choices(*args: t.Any, **kwargs: t.Any) -> t.Sequence[hikari.Comma
...
+
+    if embed.fields:
+        if (field_count := len(embed.fields)) > 25:
+            raise EmbedValidationError(f"Embed must have less than 25 fields, got {field_count}.")
+
+        for field in embed.fields:
+            if (length := len(field.name)) > 256:
+                raise EmbedValidationError(f"Embed field name must be less than 256 characters, got {length}.")
+            if (length := len(field.value)) > 1024:
+                raise EmbedValidationError(f"Embed field value must be less than 1024 characters, got {length}.")
```
Comment (Lunarmagpie) [inline toolbox/utils.py:507]: Maybe say the field name so its easier to debug?

File: toolbox/utils.py [hunk 4]
```diff
@@ -464,6 +466,55 @@ def as_command_choices(*args: t.Any, **kwargs: t.Any) -> t.Sequence[hikari.Comma
...
+def validate_embed(embed: hikari.Embed) -> hikari.Embed:
+    """Validate an embed, checking the length of all fields.
+    Parameters
+    ----------
+    embed : hikari.Embed
+        The embed to validate.
+    Raises
+    ------
+    EmbedValidationError
+        If the embed is invalid.
```
Comment (Lunarmagpie) [inline toolbox/utils.py:478]:
```suggestion
        Raised when the embed is invalid.
```

**Review by Lunarmagpie — CHANGES_REQUESTED** [review]

File: toolbox/utils.py (hunk 1 above)
Comment (Lunarmagpie) [reply toolbox/utils.py:513]: Nvm they are duplicate

Comment (parafoxia) [reply toolbox/utils.py:513]: Yeah that was a my b moment forgetting GitHub doesn't show the entire file in reviews lmao.

File: toolbox/utils.py [hunk 5]
```diff
@@ -464,6 +466,49 @@ def as_command_choices(*args: t.Any, **kwargs: t.Any) -> t.Sequence[hikari.Comma
...
+        The embed to validate.
+    Raises
+    ------
+    EmbedValidationError
+        Raised when the embed is invalid.
+    Returns
+    -------
+    hikari.Embed
+        The embed that was validated.
+    """
```
Comment (parafoxia) [inline toolbox/utils.py:483]: Sections may need spacing out, idk if it renders properly like this.

**Review by parafoxia — COMMENTED** [review]
Just something I didn't notice first time round, rest looks fine to me.

**Review by Lunarmagpie — APPROVED** [review]
Ngl I don't care enough to make the tests check for the proper error message

**Pull request merged**